client/src/contexts/ThesisCommentsProvider/ThesisCommentsProvider.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { PropsWithChildren, useEffect, useMemo, useState } from 'react'
+import { IThesis, IThesisComment } from '../../requests/responses/thesis'
+import { IThesisCommentsContext, ThesisCommentsContext } from './context'
+import { PaginationResponse } from '../../requests/responses/pagination'
+import { doRequest } from '../../requests/request'
+import { showSimpleError } from '../../utils/notification'
+
+interface IThesisCommentsProviderProps {
+  thesis: IThesis
+  commentType: 'THESIS' | 'ADVISOR'
+  limit?: number
+}
+
+const ThesisCommentsProvider = (props: PropsWithChildren<IThesisCommentsProviderProps>) => {
+  const { children, thesis, commentType, limit = 10 } = props
+
+  const [submitting, setSubmitting] = useState(false)
+  const [page, setPage] = useState(0)
+  const [comments, setComments] = useState<PaginationResponse<IThesisComment>>()
+
+  useEffect(() => {
```

Comment:
Consider optimizing the useEffect hook here. Since this effect re-fetches data every time `thesis.thesisId`, `commentType`, `page`, or `limit changes`, it might be worth looking into caching to reduce unnecessary API calls, especially if these values change frequently.

Reply:
All of these dependencies are part of the request so yes it is needed in that case as the request should always be up to date with all these states